src/name/bizna/ocarmsim/FakeMachine.java

```diff
@@ -189,15 +201,14 @@ public Architecture architecture() {
 
 	@Override
 	public Map<String, String> components() {
-		Map<String, String> ret = new HashMap<String, String>();
+		Map<String, String> ret = new HashMap<>();
```

Comment:
`new HashMap();` creates a warning because it is a raw type, `new HashMap<>();` does not because it can infer the type and is actually the same as `new HashMap<String, String>();`. The diamond operator exists since Java 7. [Question on stackoverflow.com](http://stackoverflow.com/questions/4166966/what-is-the-point-of-the-diamond-operator-in-java-7)